File: app/src/main/java/network/minter/bipwallet/tx/contract/TransactionListView.java
package network.minter.bipwallet.tx.contract;

import androidx.lifecycle.MutableLiveData;
import androidx.recyclerview.widget.RecyclerView;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
import moxy.MvpView;
import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;
import network.minter.bipwallet.internal.adapter.LoadState;
import network.minter.bipwallet.internal.mvp.ProgressView;

/**
 * minter-android-wallet. 2019
 * @author Eduard Maximovich [edward.vstock@gmail.com]
 */
public
interface TransactionListView extends MvpView, ProgressView {
    void setAdapter(RecyclerView.Adapter<?> adapter);
    void setOnRefreshListener(SwipeRefreshLayout.OnRefreshListener listener);
    void showRefreshProgress();
    void hideRefreshProgress();
    void scrollTo(int pos);
    @StateStrategyType(OneExecutionStateStrategy.class)
    void startExplorer(String hash);
    void syncProgress(MutableLiveData<LoadState> loadState);
}

File: app/src/main/java/network/minter/bipwallet/tx/contract/TransactionListView.kt
/*
 * Copyright (C) by MinterTeam. 2020
 * @link <a href="https://github.com/MinterTeam">Org Github</a>
 * @link <a href="https://github.com/edwardstock">Maintainer Github</a>
 *
 * The MIT License
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package network.minter.bipwallet.tx.contract

import androidx.lifecycle.MutableLiveData
import androidx.recyclerview.widget.RecyclerView
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout.OnRefreshListener
import moxy.MvpView
import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType
import network.minter.bipwallet.internal.adapter.LoadState
import network.minter.bipwallet.internal.mvp.ProgressView
import network.minter.bipwallet.tx.adapters.TransactionFacade
import network.minter.explorer.repo.ExplorerTransactionRepository.TxFilter

/**
 * minter-android-wallet. 2019
 * @author Eduard Maximovich [edward.vstock@gmail.com]
 */
@StateStrategyType(AddToEndSingleStrategy::class)
interface TransactionListView : MvpView, ProgressView {
    fun setAdapter(adapter: RecyclerView.Adapter<*>)
    fun setOnRefreshListener(listener: OnRefreshListener)
    fun showRefreshProgress()
    fun hideRefreshProgress()
    fun scrollTo(pos: Int)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun startExplorer(hash: String?)
    fun syncProgress(loadState: MutableLiveData<LoadState>)
    fun startDetails(tx: TransactionFacade)
    fun setFilterObserver(filterState: MutableLiveData<TxFilter>)
    fun lifecycle(state: MutableLiveData<TxFilter>, cb: (TxFilter) -> Unit)
}